templates/monitoring/kube_state_metrics_3scale_alerts.yaml

```diff
@@ -94,6 +94,24 @@ spec:
         for: 5m
         labels:
           severity: critical
+      - alert: ThreeScaleZyncPodAvailiblity
+        annotations:
+          message:  3Scale Zync has no pods in a ready state.
+          sop_url: https://github.com/RHCloudServices/integreatly-help/blob/master/sops/alerts_and_troubleshooting.md
+        expr: |
+          absent(count((kube_pod_status_ready{condition="true",namespace="redhat-rhmi-3scale",pod=~"zync-1.*"}) + on(pod) (kube_pod_container_status_running{namespace="redhat-rhmi-3scale",container="zync"}))==2)
+        for: 5m
+        labels:
+          severity: critical
+      - alert: ThreeScaleZyncDatabasePodAvailiblity
+        annotations:
+          message:  3Scale Zync-database has no pods in a ready state.
+          sop_url: https://github.com/RHCloudServices/integreatly-help/blob/master/sops/alerts_and_troubleshooting.md
+        expr: |
+          absent(sum(kube_pod_container_status_running{namespace="redhat-rhmi-3scale",pod=~"zync-database.*", container="postgresql"})==1)
```

Comment:
@austincunningham would it make sense to check the pod status rather than container?
The pod status is what determines if it's available to use (which in turn ensures all containers are ready)

Reply:
Will have a look :eyes:  in prometheus to see if it will return what we want.

Reply:
Think the reason we used kube_pod_container_status_running was it didn't return the deployment pod e.g.
kube_pod_status_ready{ namespace="redhat-rhmi-3scale", pod=~"zync-database.*" }
```
kube_pod_status_ready{condition="true",endpoint="https-main",instance="10.131.2.5:8443",job="kube-state-metrics",namespace="redhat-rhmi-3scale",pod="zync-database-1-deploy",prometheus="openshift-monitoring/k8s",service="kube-state-metrics"} | 0
-- | --
kube_pod_status_ready{condition="true",endpoint="https-main",instance="10.131.2.5:8443",job="kube-state-metrics",namespace="redhat-rhmi-3scale",pod="zync-database-1-rtqrt",prometheus="openshift-monitoring/k8s",service="kube-state-metrics"}
```

Reply:
@david-martin will use
```
absent(sum(kube_pod_status_ready{ namespace="redhat-rhmi-3scale", pod=~"zync-database.*",condition="true" })==1)
```